roles/lsaai/tasks/main.yml

```diff
@@ -1,4 +1,29 @@
 ---
+- name: "Get the list of namingContexts from all databases"
+  community.general.ldap_search:
+    server_uri: "ldapi://%2Fvar%2Frun%2Fslapd%2Fldapi"
+    dn: "olcDatabase={% raw %}{{% endraw %}{{ item }}{% raw %}}{% endraw %}mdb,cn=config"
+  loop: "{{ range(1, ldap_domains | community.general.json_query('*.ldap_db_index') | max + 1) }}"
+  failed_when: false # it's not an error if database does not exist
+  changed_when: false
+  register: ldap_naming_contexts
+  become: true
+
+- name: "Find example database that contains definitions of 'dc=my-domain,dc=com' and remove it"
+  community.general.ldap_entry:
+    server_uri: "ldapi://%2Fvar%2Frun%2Fslapd%2Fldapi"
+    dn: "olcDatabase={% raw %}{{% endraw %}{{ item.item }}{% raw %}}{% endraw %}mdb,cn=config"
+    state: absent
+    recursive: true
+    objectClass:
+      - olcDatabaseConfig
+      - olcMdbConfig
+  loop: "{{ ldap_naming_contexts.results }}"
```

Comment:
Are you sure this is working; The code loops over `ldap_naming_contexts.results` and then the `when` tests `item.results`. That suggests the data structure contains `ldap_naming_contexts.results.results.[more_nested_data]`.

Reply:
working